FHEM/14_SD_WS07.pm

```diff
@@ -119,27 +140,29 @@ SD_WS07_Parse($$)
     	$model=$model."_T";		
     } else {
     	$model=$model."_TH";		
-    	if ($hum < 10 || $hum > 99) {
-    	    return '';
+    	if ($hum > 99) {
+			Log3 $name, 3, "$iohash->{NAME}: $model ERROR - Humidity unknown ($hum)";
+			return "";
```

Comment:
Wir könnten den Humbereich vielleicht sogar noch etwas einengen. Auf der Packung des EAS800Z steht: Luftfeuchte 20 - 95 %

Reply:
Solange er nicht unter 0% anzeigt, ist alles im grünen Bereich :-)
![sd_ws07-hum](https://user-images.githubusercontent.com/26026988/32965458-d2d095fe-cbd6-11e7-8344-f3c06153fe48.png)
Clock schwankt bei mir so zwischen 481 und 517 - ist halt abhängig von der Einstellung des sens -Wertes vom CC!!0x. Ich hab halt den Wert laut Fund im Internet eingestellt "the packets are ppm modulated (distance coding) with a pulse of ~500 us " und die 16 µS sollten keine Geige spielen.

Reply:
Hier noch ein aktuelles Beispiel, das der Sensor schon unter 20 % rel. Feuchte geht:
![zwischenablage-2](https://user-images.githubusercontent.com/26026988/33234746-15e0392a-d22c-11e7-9359-5d5f02210b75.png)
Wir sollten ihm eher noch einen Offset beibringen. Bei mir liegt der Messwert immer so etwa 25 unter denen vergleichbarer Sensoren:
![zwischenablage-3](https://user-images.githubusercontent.com/26026988/33234777-a0758536-d22c-11e7-93d5-17c52ddeff4f.png)

Reply:
ich habe mal die log Ausgaben überarbeitet, und in den master commitet
```
2017.12.04 19:49:43.661 4 : sduinoD SD_WS07: P7#B72094F30, length=9
2017.12.04 19:49:43.661 5 : sduinoD SD_WS07: converted to bits: 10110111 0 010 000010010100 1111 00110000
2017.12.04 19:49:43.661 4 : sduinoD SD_WS07: model=SD_WS07_TH, id=B7, channel=3, temp=14.8, hum=48, bat=low
2017.12.04 19:49:43.661 4 : sduinoD SD_WS07: using longid=1 model=SD_WS07_TH
2017.12.04 19:49:43.661 1 : sduinoD SD_WS07: UNDEFINED sensor SD_WS07_TH_B73 detected, code P7#B72094F30
```

Jetzt fehlen noch die Attribute temperatureOffset,  humidityOffset und sanityCheck:0,1
Der sanityCheck sollte per default disabled sein

Im 14_CUL_TCM97001 wird das battery reading nur bei Änderungen aktualisiert, was haltet Ihr davon, dies auch im SD_WS07 Modul einzubauen 
```
    if ($hasbatcheck) {
      my $battery = ReadingsVal($name, "battery", "unknown");
      if ($batbit) {
        if ($battery ne  "ok") { readingsBulkUpdate($def, "battery", "ok"); }
      } else {
        if ($battery ne  "low") { readingsBulkUpdate($def, "battery", "low"); }
      }
    }
```

Reply:
Das mit dem battery update ist eine gute Idee, das spart unnötige Ausgaben in das Log des Sensors. Habe ich eben noch mit eingebaut. Die Offsets sind eingearbeitet. Sanity check per Attribut abzuschalten finde ich nicht so gut. Man könnte den Default auf einen hohen Wert setzen, dann greift das nicht sofort. Aber in der Hilfe zum Sensor müsste auf doubleMsgCheck_IDs hingewiesen werden, sonst kommt keiner darauf.
[14_SD_WS07.zip](https://github.com/RFD-FHEM/RFFHEM/files/1528753/14_SD_WS07.zip)